server/src/main/java/org/opensearch/env/Environment.java

```diff
@@ -137,6 +139,7 @@ public Environment(final Settings settings, final Path configPath, final boolean
         tmpDir = Objects.requireNonNull(tmpPath);
 
         pluginsDir = homeFile.resolve("plugins");
+        extensionsDir = homeFile.resolve("extensions");
```

Comment:
Does this work even if the folder doesn't exist? Since the feature is experimental I wonder whether we need to wrap this into an `if` (I don't think so).

Reply:
Yeah it works even if the folder isn't there, but any subsequent actions with that Path won't work without the folder.  I wrapped the areas that use this variable in `if` statements with the feature flag.